devops/providers/azure-devops/templates/infrastructure/cicd/azure-pipelines.yaml

```diff
@@ -0,0 +1,125 @@
+variables:
+
+# location of files used by pipeline
+- name: PIPELINE_SCRIPTS_ROOT
+  value: cicd
+- name: BUILD_DIR
+  value: build
+
+# dependency versions
+- name: GO_VERSION
+  value: '1.12.5'
+- name: TF_VERSION
+  value: '0.12.2'
+
+# terraform flags
+- name: TF_WARN_OUTPUT_ERRORS
+  value: 1
+- name: ARM_PROVIDER_STRICT
+  value: 'true'
+
+# test harness image location and image tagging metadata
+- name: TEST_HARNESS_BASE_IMAGE
+  value: 'msftcse/cobalt-test-base'
+- name: TEST_HARNESS_BUILD_ARTIFACT_IMAGE_PREFIX
+  value: cobalt-build-artifact
+- name: TEST_HARNESS_REPOSITORY_GIT_URL
+  value: https://github.com/microsoft/cobalt.git
+- name: TEST_HARNESS_DIR
+  value: test-harness
+
+
+stages:
+- stage: CI
+  jobs:
+
+  - job: CI
+
+
+    ### BEGIN BUILD TASKS
+
+    pool: $(AGENT_POOL)
+    workspace:
+      clean: all
+
+    steps:
+    - checkout: self
+      clean: all
+      fetchDepth: 1
+
+    - script: bash ./$(PIPELINE_SCRIPTS_ROOT)/test-harness-checkout.sh
+      displayName: Checkout Test Harness
+
+    - task: Docker@2
+      displayName: Login to Base Image ACR
+      inputs:
+        command: login
+        repository: $(BASE_IMAGE_ACR_HOST)
+        containerRegistry: $(BASE_IMAGE_ACR_SERVICE_CONNECTION_NAME)
+        
+    - script: bash ./$(PIPELINE_SCRIPTS_ROOT)/test-harness-build.sh
+      displayName: Configure Test Harness
+      env:
+        BASE_IMAGE_ACR_PASSWORD: $(DOCKER-PASSWORD)
+
+    - task: Docker@2
+      condition: always()
+      displayName: Logout From Base Image ACR
+      inputs:
+        command: logout
+        containerRegistry: $(BASE_IMAGE_ACR_SERVICE_CONNECTION_NAME)
+
+
+
+    ### BEGIN TEST TASKS
+
+    - task: AzureCLI@1
+      inputs:
+        azureSubscription: $(PIPELINE_SERVICE_CONNECTION)
+        addSpnToEnvironment: true
+        workingDirectory: $(Build.SourcesDirectory)
+        scriptLocation: scriptPath
+        scriptPath: ./$(PIPELINE_SCRIPTS_ROOT)/test-harness-run.sh
+      env:
+        ARM_ACCESS_KEY: $(ARM-ACCESS-KEY)
+        ARM_SUBSCRIPTION_ID: $(AD_SP_SUBSCRIPTION_ID)
+        ARM_TENANT_ID: $(AD_SP_TENANT_ID)
+        TF_VAR_app_dev_subscription_id: $(AD_SP_SUBSCRIPTION_ID)
+        TF_VAR_ase_name: $(ASE_NAME)
+        TF_VAR_ase_resource_group: $(ASE_RESOURCE_GROUP)
+        TF_VAR_ase_subscription_id: $(AD_SP_SUBSCRIPTION_ID)
+        TF_VAR_ase_vnet_name: $(ASE_VNET_NAME)
+        TF_VAR_remote_state_account: $(REMOTE_STATE_ACCOUNT)
+        TF_VAR_remote_state_container: $(REMOTE_STATE_CONTAINER)
+
+
+
+    ### BEGIN PUBLISH TASKS
+
+    - task: CopyFiles@2
+      inputs:
+        contents: $(PIPELINE_SCRIPTS_ROOT)/*.sh
+        sourceFolder: $(Build.SourcesDirectory)
+        targetFolder: $(Build.ArtifactStagingDirectory)
+
+    - task: CopyFiles@2
+      inputs:
+        contents: $(BUILD_DIR)/*
+        sourceFolder: $(Build.SourcesDirectory)
+        targetFolder: $(Build.ArtifactStagingDirectory)
+
+    - task: PublishBuildArtifacts@1
```

Comment:
What does this do to the files you copied in the staging dir?

Reply:
This publishes them as an artifact in VS code. Ex: https://dev.azure.com/niiodice/cobalt-ci-cd-testbed/_build/results?buildId=148&view=artifacts